rust/arrow/src/compute/kernels/take.rs

```diff
@@ -195,15 +217,21 @@ where
 
     let null_slice = null_buf.data_mut();
 
-    let new_values: Vec<T::Native> = (0..data_len)
-        .map(|i| {
-            let index = indices.value(i) as usize;
-            if array.is_null(index) {
-                bit_util::unset_bit(null_slice, i);
-            }
-            array.value(index)
-        })
-        .collect();
+    let mut new_values: Vec<T::Native> = Vec::with_capacity(data_len);
+    let mut i = 0;
+    while i < data_len {
```

Comment:
I initially implemented this just as you wrote, but I opted for the `while` since it makes a non-negligible difference in the benchmarks. 

I believe the while loop and the original `::collect::Vec<T::Native>()` version show equivalent performance – it's the while loop vs. `::collect::<Result<Vec<_>>()?` where there's difference.

Reply:
that makes sense -- it might be worth mentioning that in comments somewhere (`// use while loop for performance`)

Reply:
Added in f60e629.